ArmoniK.SDK.Client.Test/CMakeLists.txt

```diff
@@ -1,31 +1,63 @@
 cmake_minimum_required(VERSION 3.22)
 set(PROJECT_NAME ArmoniK.SDK.Client.Test)
 set(NAMESPACE ArmoniK::Sdk::Client::Test)
+set(version 0.1.0)
+
 project(${PROJECT_NAME})
+set(CMAKE_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_PREFIX}/include)
 
 set(CMAKE_CXX_STANDARD 17)
 
 SET(SOURCES_FILES_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 
 FILE(GLOB_RECURSE SRC_CLIENT_FILES ${SOURCES_FILES_DIR}/*.cpp)
 
+find_package(fmt CONFIG REQUIRED)
+find_package(ArmoniK.SDK.Common CONFIG REQUIRED)
 find_package(ArmoniK.SDK.Client CONFIG REQUIRED)
 
 add_executable(${PROJECT_NAME} ${SRC_CLIENT_FILES})
 
-target_link_libraries(${PROJECT_NAME} PRIVATE ArmoniK.SDK.Common ArmoniK.SDK.Client GTest::gtest_main)
+target_link_libraries(${PROJECT_NAME} PUBLIC fmt ArmoniK.SDK.Common ArmoniK.SDK.Client GTest::gtest_main)
 
 target_compile_definitions(${PROJECT_NAME} PUBLIC SDK_CLIENT_TEST_NAMESPACE=${NAMESPACE})
 
-setup_options(${PROJECT_NAME})
+file(READ /etc/issue ${PROJECT_NAME}_ETCISSUE_CONTENT)
+string(FIND "${${PROJECT_NAME}_ETCISSUE_CONTENT}" "Alpine" IS_ALPINE)
+
+if(MSVC)
+    target_compile_options(${PROJECT_NAME} PRIVATE /W4)
+else()
+    if(CMAKE_BUILD_TYPE MATCHES DEBUG AND IS_ALPINE EQUAL -1)
+        target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wextra -Wpedantic -fsanitize=undefined,address)
+    else ()
+        target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wextra -Wpedantic)
+    endif()
+endif()
+
+if(IS_ALPINE EQUAL -1)
+    message(STATUS "Not found Alpine distribution " ${IS_ALPINE})
+else()
+    message(STATUS "Found Alpine distribution " ${IS_ALPINE})
+endif()
+
+if(CMAKE_BUILD_TYPE MATCHES DEBUG AND IS_ALPINE EQUAL -1)
+    target_link_options(${PROJECT_NAME} PRIVATE -fsanitize=undefined,address)
+endif()
```

Comment:
Can be put in a Utils.cmake file and then this Utils.cmake can be included here with 
```include(Utils.cmake)```